src/shadowbox/server/server_access_key.ts

```diff
@@ -199,14 +207,22 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
     const metricsId = uuidv4();
     const password = params?.password ?? generatePassword();
     const encryptionMethod = params?.encryptionMethod || this.NEW_USER_ENCRYPTION_METHOD;
+    const portNumber = params?.portNumber ?? this.portForNewAccessKeys;
 
-    // Validate encryption method.
+    // Validate encryption method and port number.
     if (!isValidCipher(encryptionMethod)) {
       throw new errors.InvalidCipher(encryptionMethod);
     }
+    if (!isValidPort(portNumber)) {
+      throw new errors.InvalidPortNumber(portNumber);
+    }
+    if (portNumber !== this.portForNewAccessKeys && !(await this.isPortAvailable(portNumber))) {
+      throw new errors.PortUnavailable(portNumber);
+    }
```

Comment:
Please run this code only if there's a port parameter. We shouldn't need to validate the port if it's not specified and it will simplify the if blocks.

I think the same change would make the encryption method clearer, but no need to change that.

Reply:
Ah I see what you mean now. Done for both port and encryption method.